File: data-pipeline/de-duplication/src/main/java/org/ekstep/ep/samza/task/DeDuplicationConfig.java
package org.ekstep.ep.samza.task;


import org.apache.samza.config.Config;

import java.util.ArrayList;
import java.util.List;

public class DeDuplicationConfig {

    private final String JOB_NAME = "DeDuplication";
    private String successTopic;
    private String failedTopic;
    private String duplicateTopic;
    private String malformedTopic;
    private String defaultChannel;
    private final int dupStore;
    private int expirySeconds;
    private List<String> includedProducerIds = new ArrayList<>();


    public DeDuplicationConfig(Config config) {
        successTopic = config.get("output.success.topic.name", "telemetry.valid");
        failedTopic = config.get("output.failed.topic.name", "telemetry.failed");
        duplicateTopic = config.get("output.duplicate.topic.name", "telemetry.duplicate");
        malformedTopic = config.get("output.malformed.topic.name", "telemetry.malformed");
        defaultChannel = config.get("default.channel", "org.sunbird");
        dupStore = config.getInt("redis.database.duplicationstore.id", 7);
        expirySeconds = config.getInt("redis.database.key.expiry.seconds", 86400);
        if (!config.get("dedup.producer.include.ids", "").isEmpty()) {
            includedProducerIds = config.getList("dedup.producer.include.ids", new ArrayList<>());
        }

    }

    public String successTopic() {
        return successTopic;
    }

    public String failedTopic() {
        return failedTopic;
    }

    public String duplicateTopic() {
        return duplicateTopic;
    }

    public String malformedTopic() {
        return malformedTopic;
    }

    public String defaultChannel() {
        return defaultChannel;
    }

    public String jobName() {
        return JOB_NAME;
    }

    public int dupStore() {
        return dupStore;
    }

    public int expirySeconds() {
        return expirySeconds;
    }

    public List<String> inclusiveProducerIds() {
        return includedProducerIds;
    }

}

File: data-pipeline-flink/de-duplication/src/main/scala/org/ekstep/dp/task/DeduplicationConfig.scala
package org.ekstep.dp.task

import org.ekstep.dp.core.BaseJobConfig

import scala.collection.JavaConverters._

class DeduplicationConfig extends BaseJobConfig {

  private val serialVersionUID = 2905979434303791379L

  val dedupStore: Int = config.getInt("redis.database.duplicationstore.id")
  val cacheExpirySeconds: Int = config.getInt("redis.database.key.expiry.seconds")

  // Kafka Topics Configuration
  val kafkaInputTopic: String = config.getString("kafka.input.topic")
  val kafkaSuccessTopic: String = config.getString("kafka.output.success.topic")
  val kafkaDuplicateTopic: String = config.getString("kafka.output.duplicate.topic")
  val kafkaMalformedTopic: String = config.getString("kafka.output.malformed.topic")
  val includedProducersForDedup: List[String] = config.getStringList("dedup.producer.included.ids").asScala.toList

}
